table_bindings.go

```diff
@@ -129,16 +129,19 @@ func (t *TableMap) bindInsert(elem reflect.Value) (bindInstance, error) {
 					} else {
 						if col.DefaultValue == "" {
 							s2.WriteString(t.dbmap.Dialect.BindVar(x))
-							if col == t.version {
-								plan.versField = col.fieldName
-								plan.argFields = append(plan.argFields, versFieldConst)
-							} else {
-								plan.argFields = append(plan.argFields, col.fieldName)
-							}
-							x++
 						} else {
-							s2.WriteString(col.DefaultValue)
+							val := elem.FieldByName(col.fieldName).Interface()
+							s2.WriteString(
+								fmt.Sprintf("case when %t or %s = %s then %s else %s end",
+									val == nil, t.dbmap.Dialect.BindVar(x), getZeroValueStringForSQL(val), col.DefaultValue, t.dbmap.Dialect.BindVar(x)))
```

Comment:
Using result of `val == nil` here doesn't seem right. You will end up with a plan that has `case when true or ...` or `case when false or ...`

Reply:
Sorry. It's my wrong push. 🙇 
I'm still working on the fix...